src/main/java/apoc/nodes/Nodes.java

```diff
@@ -303,6 +304,26 @@ public boolean isDense(@Name("node") Node node) {
         }
     }
 
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.nodes.removewithlabels(['Person','Movie'],1000) | Will remove all the nodes having one of the given labels, via DETACH DELETE using apoc.periodic.iterate. Same limit used for each label. Default is 100. ")
+    public Stream<MapResult> removewithlabels(@Name("labels") List<String> labels, @Name(value = "limit",defaultValue="100")  Long limit) {
+        if (limit <= 0)
+            throw new RuntimeException("limit must be a positive integer");
+
+        Map<String,Object> results = new HashMap<String, Object>();
+        HashMap<String, Object> parameters = new HashMap<>();
+
+        for( String label: labels) {
+            if(label.trim().length()>0) {
+                String query = String.format("call apoc.periodic.iterate( 'MATCH (n:`%s`) RETURN n',  'DETACH DELETE n', {batchsize:%d,iterateList:true, parallel:true}  )", label, limit);
+                db.execute(query, parameters);
+            }
+        }
+        MapResult mapResult = new MapResult(results);
```

Comment:
you are not using the results, you should use the results of the iterate call as result for your method

inline